### PR TITLE
Azure decorator examples

### PR DESCRIPTION
**Pull request by akanchhaS**
Added the directory AzurePipelineDecorator with Snyk plugin and CLI based examples.

### CODE REVIEW

**Comment by akanchhaS** [comment]
Hi @maxjeffos - I have added few Azure decorator examples to run Snyk tasks. Could you please review the change?

File: AzurePipelineDecorator/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Note : Decorators in this repo work only with Azure DevOps 
+
+Check the official documentation [here](https://docs.microsoft.com/en-us/azure/devops/extend/develop/add-pipeline-decorator?view=azure-devops) and [here](https://docs.microsoft.com/en-us/azure/devops/extend/develop/pipeline-decorator-context?view=azure-devops) for more information about Azure Pipelines decorators.
+
+## Introduction 
+
+This project provides you guidelines on how to inject Snyk test and monitor checks post build execution of all the pipelines in the organization. 
+
+With a traditional pipeline approach a user would normally be required to add the Snyk task for every pipeline to check for any vulnerabilities. This process relies a lot on developers to remember to add this step/task in all the new and existing pipelines.
```
Comment (maxjeffos) [inline AzurePipelineDecorator/README.md:13]:
```suggestion
With a traditional pipeline approach, a user would normally be required to add the Snyk task for every pipeline to check for any vulnerabilities. This process relies a lot on developers to remember to add this step/task in all the new and existing pipelines. 
```

File: AzurePipelineDecorator/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+### Update the code 
+
+Based on the implementation of choice, you would need to update two files with relevant information: 
+
+- Update the YAML file - To include authentication, project and org name
+- Update the vss-extension.json - To include your publisher id, version and optionally -  name of extension, description 
+
+### Build and package the extensio
+
+1. First install node.js dependencies
```
Comment (maxjeffos) [inline AzurePipelineDecorator/README.md:45]: What dependencies?

File: AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+steps:
+- task: CmdLine@2
+  displayName: 'Decorator-injected Task to test Snyk using CLI'
+  inputs:
+    script: |
+      sudo npm install -g snyk
+      sudo npm install snyk-to-html -g
```
Comment (maxjeffos) [inline AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml:7]:
```suggestion
      sudo npm install -g snyk-to-html
```
to keep it the same pattern as the above line

File: AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+      sudo snyk auth $(SNYK_TOKEN)
```
Comment (maxjeffos) [inline AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml:8]: you won't need the sudo here

File: AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+      sudo snyk monitor --all-projects --org=YOUR_ORG_NAME
```
Comment (maxjeffos) [inline AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml:9]: you won't need the sudo here

File: AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+      sudo snyk test --all-projects --json | snyk-to-html -o snyk_results.html
```
Comment (maxjeffos) [inline AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml:10]: you won't need the sudo here

File: AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+steps:
+- task: CmdLine@2
+  displayName: 'Decorator-injected Task to test Snyk using CLI'
+  inputs:
+    script: |
+      sudo npm install -g snyk
```
Comment (maxjeffos) [inline AzurePipelineDecorator/cli:simple-example/snyk-pipeline-deco-cli.yml:6]: The `sudo` won't work on windows builds... maybe suggest using it only if required or something

**Review by maxjeffos — APPROVED** [review]

File: AzurePipelineDecorator/README.md (hunk 2 above)
Comment (akanchhaS) [reply AzurePipelineDecorator/README.md]: Thanks Jeff. Just updated Readme to elaborate the context

**Pull request merged**